### DISCUSSION TITLE
Migrating from Pi 4 to Pi 5

### DISCUSSION THREAD

**Discussion by meldumm** [home-assistant/operating-system#3212] [Q&A] [answered]
Hi,

I currently have a Raspberry Pi 4 with an external SSD running Home Assistant OS. Could you recommend a method for migrating to a Raspberry Pi 5? I doubt that simply connecting the SSD to a Pi 5 will work

**Comment by pedrojreis** [comment, marked as answer]
I believe the best and easiest method is just making a backup & restore.
I've done in the past across different hardware and it was smooth 😄

**Reply by agners** [reply]
Yes this.

If you use a data disk, currently we do not have a feature to "adopt" the datadisk on a new installation, but we consider adding this feature.

If you have the operating system installed on the SSD, then transferring won't work. Our operating systems builds are for a single machine only.

**Reply by andreas-noll** [reply]
I tried to do it with a backup and restore but then all the add-ons were broken and could simply not start. I could not get any logs sadly. Back to Raspi 4 it goes for now

**Reply by agners** [reply]
> I could not get any logs sadly.

Why is that? Did the frontend not load?

With the next Core release (or more specifically the latest Supervisor release which is planned to deploy to stable channel before the next Core release), it will be possible to migrate the data disk to a new system, see: https://rc.home-assistant.io/common-tasks/os/#migrating-an-external-data-disk-to-another-system

**Reply by andreas-noll** [reply]
It did load and I looked at the logs. Whenever I tried to start any of the Add-ons, it just wouldn't do that and nothing in relation to this would show up in the logs. I noticed when I started the new system for the first time that the Bluetooth integration had to be setup again because of a different address. The Add-on problem became noticeable to me first by the failing Matter integration, which was a result of the Matter Server add-on not being able to start. I did see the Error of the Matter integration in the logs but not any related to the Add-on.

**Reply by andreas-noll** [reply]
Just discovered, the Image for the Pi 5 was still 12.1 (seems to be the newest available in the Imager) but the Pi 4 was already running on 12.2. Updating on the Pi 5 after restoring the backup fixed it.

**Comment by 1966pjrl** [comment]
To me Backup and restore doesn't work.
I only restore application, SSL certificate doesn't work, many things you need to config again....
If you know any way to do it, please let me know.

Pedro Luz
